Source/WebKit/UIProcess/WebFullScreenManagerProxy.cpp

```diff
@@ -212,17 +212,28 @@ Awaitable<bool> WebFullScreenManagerProxy::enterFullScreen(IPC::Connection& conn
 #endif // ENABLE(QUICKLOOK_FULLSCREEN)
 #endif // PLATFORM(IOS_FAMILY)
 
+    IntRect transformedInitialFrame = initialFrame;
+    if (RefPtr page = m_page.get()) {
+        if (RefPtr mainFrame = page->mainFrame()) {
+            if (RefPtr mainFrameProcess = mainFrame->process()) {
+                std::optional<FloatRect> mainFrameCoordinates = co_await page->convertRectToMainFrameCoordinates(WebCore::FloatRect(initialFrame), frameID);
```

Comment:
In order for this to be 0 IPC roundtrips in the non-site-isolation case, we would want to pass in the root frame identifier and the coordinates in root frame coordinates.  Root frame is the topmost local frame in a frame tree subtree in a given process.

Also, this can be written with a lot less indentation.